core/server/master/src/main/java/alluxio/master/file/InodeSyncStream.java

```diff
@@ -769,9 +769,15 @@ private void loadMetadataForPath(LockedInodePath inodePath)
         && !inodePath.getUri().equals(mRootScheme.getPath())) {
       descendantType = DescendantType.NONE;
     }
+
+    FileSystemMasterCommonPOptions option = NO_TTL_OPTION;
+    if (Configuration.getBoolean(PropertyKey.MASTER_METADATA_SYNC_USE_CLIENT_OPTION)) {
+      option = mSyncOptions;
+    }
+
     LoadMetadataContext ctx = LoadMetadataContext.mergeFrom(
         LoadMetadataPOptions.newBuilder()
-            .setCommonOptions(NO_TTL_OPTION)
```

Comment:
I think by default sync will just create the file with the same TTL option for the path (and inherit from the parent). Let's not add another option. @maobaolong do you see a use case where you want the sync-ed path to NOT have inherited TTL?

Reply:
here you replace a default option with the load metadata option passed from the caller. If LoadDescendantPType == ONE, will this result in you ending up loading all files/directories recursively as if LoadDescendantPType is set to true, because you always load "the next level"? @jiacheliu3 I'm not very familiar with the code path may be we three can discuss offline.